server/src/unifyfs_request_manager.c

```diff
@@ -1042,57 +1123,37 @@ static int rm_process_remote_chunk_responses(reqmgr_thrd_t* thrd_ctrl)
                     ret = rc;
                 }
             }
-        }
-    }
+        } else if ((req->num_remote_reads == 0) &&
```

Comment:
I'm confused about this case (i.e., `num_remote_reads == 0`). Can you explain the scenario when this occurs? That field is a count of servers who need to respond (including the local server if applicable). Any request should have at least one server, right?

Reply:
This happens in the case where a client issues a read for a segment of a file that does not exist, e.g., the read segment falls fully within a hole.

I was at first trying to modify things to have the server notify the client right away by adding a new field to the rpc, but that ran into snags.  Also I realized that this required the server to determine before it returns from the read/mread RPC whether it will find any data.  While we do that now, I can imagine we may move that out of the critical path later.  So it felt cleaner to have the client/server go through the normal flow where the client waits for read replies to come back from the server through shared memory.  In this case the server just informs the client that nothing is coming (repies=0 and flag=COMPLETE).

Reply:
ah, ok. makes sense now